library/alloc/src/collections/btree/node.rs

```diff
@@ -9,11 +9,8 @@
 // struct Node<K, V, height: usize> {
 //     keys: [K; 2 * B - 1],
 //     vals: [V; 2 * B - 1],
-//     edges: if height > 0 {
-//         [Box<Node<K, V, height - 1>>; 2 * B]
-//     } else { () },
-//     parent: Option<NonNull<Node<K, V, height + 1>>>,
-//     parent_idx: u16,
+//     edges: [if height > 0 { Box<Node<K, V, height - 1>> } else { () }; 2 * B],
+//     parent: Option<(NonNull<Node<K, V, height + 1>>, u16)>,
```

Comment:
`parent` is unrelated to leaf edges, but the fact that `parent_idx` is not part of the option is not "the ideal", it's probably the only way we can have the compiler merge the two u16 fields. I was going to undo that but it's in here now,